### ISSUE TITLE
Acquiring channel multiple times within one instruction should error

### ISSUE THREAD

**Issue by lcapelluto** [Qiskit/qiskit-terra#3741]
### Information

- **Qiskit Terra version**:
- **Python version**:
- **Operating system**:

### What is the current behavior?

I can acquire the same channel within the same instruction:

Acquire(5)([AcquireChannel(0), AcquireChannel(0)], [MemorySlot(0), MemorySlot(1)])

### What is the expected behavior?

This should raise an error

### Suggested solutions

Could be validated at assemble time

**Comment by taalexander** [comment]
This should be resolved by #3574.

**Comment by SooluThomas** [comment]
Looks like this has bug has been fixed.
```
Acquire(5)([AcquireChannel(0), AcquireChannel(0)], [MemorySlot(0), MemorySlot(1)])
```
raises a PulseError (`PulseError: 'Overlap on channel AcquireChannel(0) over time range [0, 5]'`)
